src/game/physics.rs

```diff
@@ -0,0 +1,50 @@
+//! Run a very simple physics simulation.
+
+use bevy::{
+    ecs::component::{ComponentHooks, StorageType},
+    prelude::*,
+};
+
+pub(super) fn plugin(app: &mut App) {
+    // `FixedUpdate` runs before `Update`, so the physics simulation is advanced
+    // before the player's visual representation is updated.
+    app.add_systems(FixedUpdate, advance_physics);
+}
+
+/// How many units per second the player should move.
+#[derive(Debug, Component, Clone, Copy, PartialEq, Default, Deref, DerefMut)]
+pub(crate) struct Velocity(pub(crate) Vec3);
+
+/// The actual transform of the player in the physics simulation.
+/// This is separate from the `Transform`, which is merely a visual
+/// representation.
+#[derive(Debug, Clone, Copy, PartialEq, Default, Deref, DerefMut)]
+pub(crate) struct PhysicalTransform(pub(crate) Transform);
+
+/// Make sure that this component is always initialized
+/// with the same value as the `Transform`
+impl Component for PhysicalTransform {
+    const STORAGE_TYPE: StorageType = StorageType::Table;
+
+    fn register_component_hooks(hooks: &mut ComponentHooks) {
+        hooks.on_add(|mut world, entity, _component_id| {
+            let rendered_transform = *world.get::<Transform>(entity).unwrap();
+            let mut physical_transform = world.get_mut::<PhysicalTransform>(entity).unwrap();
+            physical_transform.0 = rendered_transform;
+        });
+    }
+}
+
+/// Advance the physics simulation by one fixed timestep. This may run zero or
+/// multiple times per frame.
+///
+/// Note that since this runs in `FixedUpdate`, `Res<Time>` would be
```

Comment:
I feel like being explicit within the system that its timestep will be fixed is not idiomatic. Same opinion for other instances of this in the PR ofc.

Reply:
I disagree on that, as the user might not be familiar with how time steps work, so they might not realize that this `Time` is different from the `Time` they're used to. @MiniaczQ, can you do a tie-breaker? I'll accept it if that's the majority opinion.

Reply:
Imo systems should be designed with schedule in mind, I'm not a huge fan of `Time<()>`

Reply:
[`a4c1565` (#70)](https://github.com/TheBevyFlock/bevy-template/pull/70/commits/a4c1565bd7d5e0fdb41255e19ea148d77c0c24b4)